File: server/src/MerchWebsite.API/Models/DTOs/CreateOrderDto.cs
// ~/Projects/MerchWebsite/server/src/MerchWebsite.API/Models/DTOs/CreateOrderDto.cs
using System.ComponentModel.DataAnnotations;

namespace MerchWebsite.API.Models.DTOs
{
    // DTO for the request body when creating an order
    public class CreateOrderDto
    {
        // We might not need anything specific here if we fetch cart from DB
        // but we DO need shipping address.

        // Could add a flag: public bool SaveAddress { get; set; }

        [Required]
        public string ShippingAddress_FullName { get; set; } = string.Empty;
        [Required]
        public string ShippingAddress_AddressLine1 { get; set; } = string.Empty;
        public string? ShippingAddress_AddressLine2 { get; set; }
        [Required]
        public string ShippingAddress_City { get; set; } = string.Empty;
        [Required]
        public string ShippingAddress_PostalCode { get; set; } = string.Empty;
        [Required]
        public string ShippingAddress_Country { get; set; } = string.Empty;
    }
}

File: client/src/app/models/order/create-order.dto.ts
// client/src/app/models/order/create-order.dto.ts
// Matches backend CreateOrderDto
export interface CreateOrderDto {
  shippingAddress_FullName: string;
  shippingAddress_AddressLine1: string;
  shippingAddress_AddressLine2: string | null; // Match null possibility
  shippingAddress_City: string;
  shippingAddress_PostalCode: string;
  shippingAddress_Country: string;
  // Add SaveAddress flag later if needed
}
